proto/substrait/relations.proto

```diff
@@ -246,3 +246,366 @@ message Rel {
     CrossRel cross = 12;
   }
 }
+
```

Comment:
it probably makes sense to rename this file to something more apt to the content inside.

Reply:
The best I came up with was `algebra.proto` 🤷🏻.